can-compute-stream.js

```diff
@@ -32,56 +43,83 @@ var computeStream = function () {
-			return lastValue;
-		},
-		set: function (val) {
-			return val;
-		},
-		on: function (updated) {
-			streamHandler = function (val) {
-				lastValue = val;
-				updated();
-			};
-			valueStream.onValue(streamHandler);
-		},
-		off: function () {
-			valueStream.offValue(streamHandler);
-		}
-	});
+/*
+ * Returns a single stream for a property on an {observable}
+ */
+computeStream.toStreamFromProperty = function( obs, propName ) {
+	return computeStream.toStreamFromCompute(compute(obs, propName));
+};
+
+/*
+ * Returns a single stream for a specific event on an {observable} property
+ */
+computeStream.toStreamFromEvent = function() {
+	var obs = arguments[0];
+	var eventName, propName;
+
 
-	return valueCompute;
+	if(arguments.length === 2) {
+		//.toStreamFromEvent(obs, event);
+
+		eventName = arguments[1];
+		
+        return Kefir.stream(function (emitter) {
+			var handler = function(ev){
+                var clone = assign({}, ev);
+                ev.args = Array.prototype.slice.call(arguments, 1);
+                emitter.emit(clone);
+            };
+
+			canEvent.addEventListener.call(obs, eventName, handler);
+
+			return function(){
+				canEvent.removeEventListener.call(obs, eventName, handler);
+            };
+        });
+    } else {
+		//.toStreamFromEvent(obs, propName, event);
+		propName = arguments[1];
+		eventName = arguments[2];
+
+		var propValueStream = computeStream.toStreamFromProperty(obs, propName);
+
+		return Kefir.stream(function (emitter) {
+            var handler = function(ev){
+                var clone = assign({}, ev);
+                ev.args = Array.prototype.slice.call(arguments, 1);
+                emitter.emit(clone);
+            };
+            var curValue;
+
+            propValueStream.onValue(function(value){
+                if(curValue) {
+                    canEvent.removeEventListener.call(curValue, eventName, handler);
+                }
+                if(value) {
+                    canEvent.addEventListener.call(value, eventName, handler);
+                }
+                curValue = value;
+            });
+
+
+            return function(){
+                if(curValue) {
+                    canEvent.removeEventListener.call(curValue, eventName, handler);
+                }
+            };
+        });
+    }
 };
 
-define.behaviors.push('stream');
-
-var oldExtensions = define.extensions;
-
-define.extensions = function (objPrototype, prop, definition) {
-	if (isArray(definition.stream)) {
-		return assign({
-			value: function () {
-				var map = this;
-				var computes = definition.stream
-					.map(function (arg) {
-						return typeof arg === 'string' ? compute(map, arg) : arg;
-					});
-				return computeStream.asCompute.apply(this, computes);
-			}
-		}, define.types.compute);
-	} else {
-		return oldExtensions.apply(this, arguments);
-	}
+/*
+ * Takes multiple streams and returns a single stream
+ */
+computeStream.toSingleStream = function() {
```

Comment:
I am using this in a test case that merges multiple streams into a single stream.